src/test/java/com/gluonhq/richtext/undo/CmdManagerTests.java

```diff
@@ -52,6 +54,47 @@ public void redoRules() {
 
     }
 
+    private CommandManager<StringBuilder> commander;
+    private AtomicBoolean emptyUndo;
+    private AtomicBoolean emptyRedo;
+
+    @Test
+    @DisplayName("Passes multiple redo rules")
+    public void multipleRedoRules() {
```

Comment:
This tests needs to be broken into at least 3 different tests:

1. Execute `TestCommand` and assert the size of both the stacks using the new methods.
2. Execute `TestCommand` followed by a `undo`. Assert the size of stacks.
3. Execute `TestCommand`. Fire `undo` followed by a `redo`. Assert the size of stacks.

Reply:
Do you think I could simply add the asserts to the three existing tests:

```
public void executionRules() {
...
       Assertions.assertEquals( 1,  commander.undoStack.size());
       Assertions.assertEquals( 0,  commander.redoStack.size());
+       Assertions.assertTrue(commander.isUndoStackEmpty());
+       Assertions.assertTrue(commander.isRedoStackEmpty());
       Assertions.assertEquals( "Text-redo",  commander.context.toString());
   }
```

Reply:
Yup, that's exactly what I was suggesting. Instead of asserting against `emptyUndo` and `emptyRedo` fields, we can directly assert the new methods which returns the state of the underlying stacks.